pkg/ibmvpc/parser.go

```diff
@@ -613,13 +633,51 @@ func getNACLconfig(rc *datamodel.ResourcesContainerModel,
 	return nil
 }
 
+func getTgwMap(c *datamodel.ResourcesContainerModel) map[string]*datamodel.TransitGateway {
+	tgwIDToTgw := map[string]*datamodel.TransitGateway{}
+	for _, tgw := range c.TransitGatewayList {
+		tgwIDToTgw[*tgw.Crn] = tgw
+	}
+	return tgwIDToTgw
+}
+
 func getTgwObjects(c *datamodel.ResourcesContainerModel,
-	res map[string]*vpcmodel.VPCConfig) map[string]*TransitGateway {
+	res map[string]*vpcmodel.VPCConfig, resourceGroup string, regions []string) map[string]*TransitGateway {
 	tgwMap := map[string]*TransitGateway{} // collect all tgw resources
+	tgwIDToTgw := map[string]*datamodel.TransitGateway{}
+	if resourceGroup != "" || len(regions) > 0 {
+		tgwIDToTgw = getTgwMap(c)
+	}
+
 	for _, tgwConn := range c.TransitConnectionList {
 		tgwUID := *tgwConn.TransitGateway.Crn
 		tgwName := *tgwConn.TransitGateway.Name
 		vpcUID := *tgwConn.NetworkID
+
+		// filtering by resourceGroup
+		if resourceGroup != "" {
+			if tgw, ok := tgwIDToTgw[tgwUID]; ok {
+				if tgw == nil || *tgw.ResourceGroup.ID != resourceGroup {
+					continue
+				}
+			} else {
+				fmt.Printf("warning: ignoring tgw with unknown resource-group, tgwID: %s\n", tgwUID)
+				tgwIDToTgw[tgwUID] = nil // to avoid having this tgw's same warning issued again from another transitConnection
+				continue
+			}
+		}
+		// filtering by region
+		if len(regions) > 0 {
+			if tgw, ok := tgwIDToTgw[tgwUID]; ok {
+				if tgw == nil || !slices.Contains(regions, *tgw.Location) {
+					continue
+				}
+			} else {
+				fmt.Printf("warning: ignoring tgw with unknown region, tgwID: %s\n", tgwUID)
+				tgwIDToTgw[tgwUID] = nil // to avoid having this tgw's same warning issued again from another transitConnection
+				continue
+			}
+		}
```

Comment:
1. As Ziv suggested, better have a `map[string]bool` that keeps a set of TGWs to skip.. + avoid current code duplication for warnings handling 
2. missing tests for filtering of TGW by region as well